src/ol/Overlay.js

```diff
@@ -180,18 +167,11 @@ class Overlay extends BaseObject {
     this.element.style.position = 'absolute';
     this.element.style.pointerEvents = 'auto';
 
-    let autoPan = options.autoPan;
-    if (autoPan && 'object' !== typeof autoPan) {
-      autoPan = {
-        animation: options.autoPanAnimation,
-        margin: options.autoPanMargin,
-      };
-    }
     /**
      * @protected
-     * @type {PanIntoViewOptions|false}
+     * @type {PanIntoViewOptions|undefined}
```

Comment:
The `|undefined` here should be redundant, because you already set the `autoPan` option in `Options` optional.

But looking at the code, I see that any truthy value will enable auto pan with defaults, so it might make sense to set `boolean` as allowed type in addition to `PanIntoViewOptions` in the `Options`.

Reply:
In a non-strict context `| undefined` can be omitted because `null` / `undefined` is implicitly allowed.
If autoPan were a local variable its type could be inferred from context.
Class member types cannot be inferred because the variable could be set to something else in a different function.

Without `| undefined` and `strict` enabled you will get an error:
> Type 'PanIntoViewOptions | undefined' is not assignable to type 'PanIntoViewOptions'.
> Type 'undefined' is not assignable to type 'PanIntoViewOptions'.ts(2322)

Makes sense to allow `true` for auto-pan with default options.